src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -82,11 +82,20 @@ public static class ExpressionMetadataProvider
 
             object modelAccessor(object container)
             {
-                var compiledExpression = CachedExpressionCompiler.Process(expression);
-                Debug.Assert(compiledExpression != null);
+                var model = (TModel)container;
+                var cachedFunc = CachedExpressionCompiler.Process(expression);
+                if (cachedFunc != null)
+                {
+                    return cachedFunc(model);
```

Comment:
This line makes an incorrect assumption. For example `Process_ConstantMember_WithNullConstant(...)` demonstrates the `CachedExpressionCompiler` does not avoid all NREs.

Reply:
Ahh, that was a bad test. It returns a null `func` which is separately covered by `Process_ConstantMemberAccess_ReturnsNull`

Reply:
Guess we're missing a test as well as having a problem here. `CompileCapturedConstant(...)` will return a delegate that `throw`s an NRE when the captured local is `null`.

Reply:
Verified that `CompileCapturedConstant` only works for an expression like `var foo = "Hello"; RunExpression(m => foo)` `RunExpression(m => foo.Length)` returns a null, which is now covered by an additional test `Process_MemberAccessOnCapturedVariable_ReturnsNull`

Reply:
Cool. Please fix the comment at the start of `CompileCapturedConstant(...)`

Reply:
thx